src/joint_types.jl

```diff
@@ -131,6 +131,80 @@ function velocity_to_configuration_derivative!(q̇::AbstractVector, jt::Quaterni
     nothing
 end
 
+
+"""
+$(SIGNATURES)
+
+Copy the matrix ``source`` to a block of the same size in ``destination``
+spanning the given given range of rows and columns. In Julia v0.7 and above we
+should be able to eliminate this function in favor of:
+
+```julia
+destination[dest_rows, dest_cols] .= source
+```
+
+since the resulting view will be elided.
+"""
+function _copyto_offset!(destination::AbstractMatrix, source::AbstractMatrix, dest_rows::UnitRange{Int} = 1:size(source, 1), dest_cols::UnitRange{Int} = 1:size(source, 2))
```

Comment:
Wouldn't `Compat.copyto!` with `CartesianRange`s do the trick?

```
  copy!(dest, Rdest::CartesianRange, src, Rsrc::CartesianRange) -> dest

  Copy the block of src in the range of Rsrc to the block of dest in the range
  of Rdest. The sizes of the two regions must match.
```

Reply:
But make sure that doesn't allocate; I seem to recall a performance issue with that method in 0.4; maybe that has been fixed since.

Reply:
Nice. That works and doesn't allocate on v0.6

Reply:
no longer needed at all